src/main/java/com/chatapp/vartalaab/service/MessageService.java

```diff
@@ -64,10 +79,32 @@ public void saveMessagesToCache(MessageDto messageDto){
         offlineMessageIdsRepository.save(tempOfileMessageIds);
     }
 
-    public void forwardTheMessageToReceiver(String messageReceiver, TextMessage message) throws IOException {
+    public void forwardTheMessageToReceiver(String messageReceiver, TextMessage message, boolean isAck) throws IOException {
         for(String webSocketSessionId: userSessionService.getUserWebSocketSessionIds(messageReceiver)){
             sessionMap.get(webSocketSessionId).sendMessage(message);
-            //Todo: save msg to MongoDB
+        }
+        if(!isAck) {
```

Comment:
what does this ack represents ?

Reply:
It's to identify if it's an acknowledgement message like "sent", "received", "read"